indico/modules/events/registration/__init__.py

```diff
@@ -55,8 +55,10 @@ def _get_open_regforms(event):
 @template_hook('conference-home-info')
 def _inject_regform_announcement(event, **kwargs):
     regforms = _get_open_regforms(event)
+    user_has_registered = session.user and any(regform.get_registration(user=session.user) for regform in regforms)
```

Comment:
user_has_registered = session.user and Registration.find(user=session.user).with_parent(regform).count()